util/src/main/java/tech/pegasys/artemis/util/config/ArtemisConfigurationBuilder.java

```diff
@@ -343,7 +363,12 @@ public ArtemisConfiguration build() {
         restApiInterface);
   }
 
-  private <T> T getEffectiveValue(
+  private <T> T getEffectiveValue(final T explicitValue, final Supplier<T> predefinedNetworkValue) {
```

Comment:
nit: the naming of these two methods is a bit confusing since the getEffectiveValueOptional is currently the one that possibly returns null whereas getEffectiveValue wraps it in an optional

Reply:
The different is actually about whether the default value is `Optional` or not. I can't think of a particularly clear name but have changed to `getOrDefault` and `getOrOptionalDefault` to match the `Map.getOrDefault` method since that's essentially what it does. Ideally they'd have the same name but you can't have method signatures where only the generics vary.